crates/worker/src/partition_processor_manager/mod.rs

```diff
@@ -251,6 +271,9 @@ impl PartitionProcessorManager {
                 Some(partition_processor_rpc) = self.incoming_partition_processor_rpc.next() => {
                     self.on_partition_processor_rpc(partition_processor_rpc);
                 }
+                Some(result) = self.snapshot_export_tasks.next() => {
+                    self.on_snapshot_task_result(result.expect("export task ok??"));
+                }
```

Comment:
I don't quite get why `.next()` returns a `Result<Result<...>>` here - `FuturesUnordered` talks about using `poll_next()` but I am getting:

> error[E0599]: no method named `poll_next` found for struct `FuturesUnordered` in the current scope

This I believe is because my Item type isn't `Pin` - but I also tried to wrap the handle in a `Pin<Box<...>>` and that didn't solve it. Educate me please? :-)

Reply:
Mystery solved: this is because `TaskHandle` returns a `Result` of the return type, which is itself `Result`.